core/controllers/domain_objects_validator.py

```diff
@@ -178,3 +178,14 @@ def validate_aggregated_stats(aggregated_stats):
                 raise base.BaseHandler.InvalidInputException(
                     '%s not in state stats mapping of %s in aggregated '
                     'stats dict.' % (state_stats_property, state_name))
+
+
+def validate_params_dict(params):
+    """validates params dict
+
+    Args:
+        params: dict. Data that needs to be validated.
+    """
+    for key in params.keys():
+        if not isinstance(key, str):
+            raise Exception('Excepted str, received %s' % key)
```

Comment:
@Nik-09 After adding print logs I found that though I am passing non-string type data this condition still remains false.